components/ILIAS/Certificate/classes/Setup/class.ilCertificateDatabaseUpdateSteps.php

```diff
@@ -77,4 +77,18 @@ public function step_5(): void
             $this->db->addIndex('il_cert_user_cert', ['background_image_path', 'currently_active'], 'i7');
         }
     }
+
+    public function step_6(): void
+    {
+        if (
+            $this->db->tableExists('il_cert_user_cert')
+            && !$this->db->tableColumnExists('il_cert_user_cert', 'certificate_id')
+        ) {
+            $this->db->addTableColumn('il_cert_user_cert', 'certificate_id', [
```

Comment:
One additional idea: Let's add a migration step in the setup agent of the certificate component. This migration adds a unique ID for all existing records in `il_cert_user_cert` (please check whether we could use the `uuid` dependencies of if we need a local `uuid` generator in the migration).

By doing this, we could add a `Unique Constraint` for column `certificate_id` and set `notnull` to `true` if the migration step is successfully finished.